src/unistd.rs

```diff
@@ -539,6 +539,23 @@ pub fn symlinkat<P1: ?Sized + NixPath, P2: ?Sized + NixPath>(
     Errno::result(res).map(drop)
 }
 
+// Double the buffer capacity up to limit. In case it already has
+// reached the limit, return Errno::ERANGE.
+fn reserve_buffer_size<T>(buf: &mut Vec<T>, limit: usize) -> Result<()> {
+    use std::cmp::min;
+
+    if buf.len() == limit {
+        return Err(Error::Sys(Errno::ERANGE))
+    }
+
+    unsafe { buf.set_len(buf.capacity()) };
```

Comment:
No, absolutely not.  This is very unsafe and totally unnecessary.  Did you mean to commit this line?